RecoParticleFlow/PFClusterProducerAlpaka/plugins/alpaka/PFClusterProducerKernel.dev.cc

```diff
@@ -0,0 +1,2224 @@
+                                  int* __restrict__ globalRechits) const {
+      const int nRH = pfRecHits.size();
+      hcalFastCluster_serial(acc,
+                             pfClusParams,
+                             nRH,
+                             pfRecHits,
+                             clusteringVars,
+                             clusterView,
+                             fracView,
+                             globalClusterPos,
+                             globalPrevClusterPos,
+                             globalClusterEnergy,
+                             globalRhFracSum,
+                             globalSeeds,
+                             globalRechits);
+    }
+  };
+
+  PFClusterProducerKernel::PFClusterProducerKernel(cms::alpakatools::device_buffer<Device, uint32_t>&& buffer1,
+                                                   //cms::alpakatools::host_buffer<uint32_t>&& buffer2,
+                                                   cms::alpakatools::device_buffer<Device, Position4[]>&& buffer3,
+                                                   cms::alpakatools::device_buffer<Device, Position4[]>&& buffer4,
+                                                   cms::alpakatools::device_buffer<Device, float[]>&& buffer5,
+                                                   cms::alpakatools::device_buffer<Device, float[]>&& buffer6,
+                                                   cms::alpakatools::device_buffer<Device, int[]>&& buffer7,
+                                                   cms::alpakatools::device_buffer<Device, int[]>&& buffer8)
+      : nSeeds(std::move(buffer1)),
+        //nTopos(std::move(buffer2)),
+        globalClusterPos(std::move(buffer3)),
+        globalPrevClusterPos(std::move(buffer4)),
+        globalClusterEnergy(std::move(buffer5)),
+        globalRhFracSum(std::move(buffer6)),
+        globalSeeds(std::move(buffer7)),
+        globalRechits(std::move(buffer8)) {}
+
+  PFClusterProducerKernel PFClusterProducerKernel::Construct(Queue& queue,
+                                                             const reco::PFRecHitHostCollection& pfRecHits) {
+    const int nRH = pfRecHits->size();
+
+    return PFClusterProducerKernel{cms::alpakatools::make_device_buffer<uint32_t>(queue),
+                                   //cms::alpakatools::make_host_buffer<uint32_t>(queue),
+                                   cms::alpakatools::make_device_buffer<Position4[]>(queue, 3000 * 1000),
+                                   cms::alpakatools::make_device_buffer<Position4[]>(queue, 3000 * 1000),
+                                   cms::alpakatools::make_device_buffer<float[]>(queue, 1),
+                                   cms::alpakatools::make_device_buffer<float[]>(queue, 1),
+                                   cms::alpakatools::make_device_buffer<int[]>(queue, 1),
+                                   cms::alpakatools::make_device_buffer<int[]>(queue, 1)};
```

Comment:
Rather than allocating 5 4-byte variables separately, how about crafting a `struct` to hold them and allocate an instance of that struct once?